#152109581 Validate Address/Payment Details

fidelisojeah opened the pull request:
#### What does this PR do?
- Ensures there is a validation before the address form is submitted

#### How should this be manually tested?
- By trying to add an address and entering invalid details

#### What are the relevant pivotal tracker stories?
- 152109581

#### Screenshots (if appropriate)
<img width="1088" alt="screen shot 2017-10-26 at 5 55 30 pm" src="https://user-images.githubusercontent.com/11561738/32066185-ea9c75e8-ba76-11e7-81bc-ca10c3f8d875.png">

#### Questions:
- How do you validate an address is an address?
- How do you validate name?

fidelisojeah committed:
fix(bug-fix): validate address details

- ensure address details are verified on save
>- Phone number has a verification mechanism
- An extra validation field message was added to the address schema on startup

[Fixes #152109581]

femisenjobi reviewed (changes requested):
When I enter an invalid number, and click `Save and Continue` the phone validation happends but other fields are cleared. Let's fix that and we are good to go.

<img width="805" alt="screen shot 2017-10-31 at 11 04 20 am" src="https://user-images.githubusercontent.com/17007838/32218477-46a48454-be2b-11e7-93da-b5797dd59ea8.png">

fidelisojeah committed:
fix(bug-fix): validate address details

- Clears an error where vlaues in address field are cleared if an invalid phone number is entered (after validation)

[Fixes #152109581]

fidelisojeah commented:
This has now been fixed 

<img width="974" alt="screen shot 2017-10-31 at 11 32 52 am" src="https://user-images.githubusercontent.com/11561738/32219878-f007261a-be2f-11e7-80f9-478e8d60999d.png">
